src/compiler/parser.ts

```diff
@@ -1001,10 +1001,10 @@ module ts {
             switch (parsingContext) {
                 case ParsingContext.SourceElements:
                 case ParsingContext.ModuleElements:
-                    return isSourceElement(inErrorRecovery);
+                    return !(token === SyntaxKind.SemicolonToken && inErrorRecovery) && isModuleElement();

```

Comment:
No, I much prefer it this way. It tells me exacly what is going on and I don't have to navigate to a "helper" function to figure it out.